### ISSUE TITLE
[no-unused-vars] False positive with JSX

### ISSUE THREAD

**Issue by bradzacher** [bradzacher/eslint-plugin-typescript#283]
**Repro**

```JSON
{
  "rules": {
    "typescript/no-unused-vars": "error"
  },
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    }
  }
}
```

```TS
const Foo = function () {}
export function render() {
    return <Foo />;
}
```

**Expected Result**
a-okay

**Actual Result**
`'Foo' is assigned a value but never used.`

**Versions**

| package                    | version |
| -------------------------- | ------- |
| `eslint-plugin-typescript` | `1.0.0-rc.2` |
| `typescript-eslint-parser` | `included` |
| `typescript`               | `3.1.1` |

**Comment by j-f1** [comment]
This can be fixed by the [`react/jsx-uses-vars`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-uses-vars.md) rule.

**Comment by bradzacher** [comment]
Considering `jsx` is natively supported by typescript, is this something we want to pull into our plugin?

**Comment by j-f1** [comment]
I’d say no, since the React plugin already has it covered and there’s no reason to duplicate work.
